### ISSUE TITLE
Coordination member_id should be string according to tooz source

### ISSUE THREAD

**Issue by cristicalin** [gnocchixyz/gnocchi#130]
https://github.com/gnocchixyz/gnocchi/commit/a8eb25e04cdcec26161ae31e65a26b8a3f326daf introduced a change to align with tooz docs but according to the source code tooz expects the member_id to be a string (https://github.com/openstack/tooz/blame/master/tooz/coordination.py#L754-L755).

```
    :param member_id: the id of the member
    :type member_id: str
```

When using bytes as per tooz documentation the library fails to initialise due to a utf8 conversion.

Some drivers of tooz are still not ready, memcache text protocol has an issue and so does the consul driver which still expects a string which it decodes to form a session_name.

I propose that the above change be rolled back until tooz fully updates its drivers.

**Comment by jd** [comment]
IIRC it really wants bytes in Python 3. Do you have a backtrace so we can fix tooz?

**Comment by cristicalin** [comment]
Yes, indeed I am running it with python 2.7, I have not tested with python3.

Here is a log for the memcache issue (it does not throw a trace).
```
2017-06-19 15:39:24,988 [67] ERROR gnocchi.utils: Unable to start coordinator: Non-ASCII key: ''_TOOZ_MEMBER_2\x85\xee\x84Z\xeeE\xb1\xa2\xafb\xd2\x97\x94\x993''
2017-06-19 15:39:25,990 [67] ERROR gnocchi.utils: Unable to start coordinator: Non-ASCII key: ''_TOOZ_MEMBER_2\x85\xee\x84Z\xeeE\xb1\xa2\xafb\xd2\x97\x94\x993''
2017-06-19 15:39:27,992 [67] ERROR gnocchi.utils: Unable to start coordinator: Non-ASCII key: ''_TOOZ_MEMBER_2\x85\xee\x84Z\xeeE\xb1\xa2\xafb\xd2\x97\x94\x993''
2017-06-19 15:39:31,996 [67] ERROR gnocchi.utils: Unable to start coordinator: Non-ASCII key: ''_TOOZ_MEMBER_2\x85\xee\x84Z\xeeE\xb1\xa2\xafb\xd2\x97\x94\x993''
2017-06-19 15:39:40,002 [67] ERROR gnocchi.utils: Unable to start coordinator: Non-ASCII key: ''_TOOZ_MEMBER_2\x85\xee\x84Z\xeeE\xb1\xa2\xafb\xd2\x97\x94\x993''
2017-06-19 15:39:56,019 [67] ERROR gnocchi.utils: Unable to start coordinator: Non-ASCII key: ''_TOOZ_MEMBER_2\x85\xee\x84Z\xeeE\xb1\xa2\xafb\xd2\x97\x94\x993''
2017-06-19 15:40:28,047 [67] ERROR gnocchi.utils: Unable to start coordinator: Non-ASCII key: ''_TOOZ_MEMBER_2\x85\xee\x84Z\xeeE\xb1\xa2\xafb\xd2\x97\x94\x993''
```

For the consul part the trace points to this: https://github.com/openstack/tooz/blob/master/tooz/drivers/consul.py#L118
I don't have a trace on hand but here is a small repro:
```
Python 2.7.12 (default, Nov 19 2016, 06:48:10)
[GCC 5.4.0 20160609] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import uuid
>>> from oslo_utils import encodeutils
>>> my_id = uuid.uuid4().bytes
>>> _session_name = encodeutils.safe_decode(my_id)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/gnocchi/lib/python2.7/site-packages/oslo_utils/encodeutils.py", line 63, in safe_decode
    return text.decode('utf-8', errors)
  File "/opt/gnocchi/lib/python2.7/encodings/utf_8.py", line 16, in decode
    return codecs.utf_8_decode(input, errors, True)
UnicodeDecodeError: 'utf8' codec can't decode byte 0xed in position 3: invalid continuation byte
```

**Comment by jd** [comment]
Ok, so the problem is actually not bytes, but that they want ASCII-only bytes. Damn. I've updated tooz doc https://review.openstack.org/475826 and fixed it in #133.

**Comment by ParsaX** [comment]
Good, thanks mate. By changing the coordination.py and adding OS_AUTH_TYPE to my admin-openrc, I’m getting rid of 401. What is the next step that I should take?

Thanks sincerely,

Parsa

Parsa Samet
parsasamet@icloud.com

**Comment by jd** [comment]
Fixed and backported to `stable/4.0`.
